backend/src/middlewares/check-user-has-permit-to-playlist/check-user-has-permit-to-playlist.middleware.ts

```diff
@@ -9,12 +9,19 @@ const checkUserHasPermitToPlaylist = (): RequestHandler => {
       return next();
     }
 
-    const playlistId = Number(req.params.id);
+    const playlistId = Number(req.params.id ?? req.params.playlistId);
     const userId = req.user?.id;
-    const playlist = await playlistService.getById(playlistId);
 
-    if (playlist.status === PlaylistStatus.PUBLISHED || playlist.userId === userId) {
-      return next();
+    try {
+      const playlist = await playlistService.getById(playlistId);
+      if (playlist.status === PlaylistStatus.PUBLISHED || playlist.userId === userId) {
+        return next();
+      }
+    } catch (e) {
+      next(new HttpError({
+        status: HttpCode.NOT_FOUND,
+        message: e.message,
+      }));
     }
```

Comment:
```suggestion
      const playlist = await playlistService.getById(playlistId);
      if (playlist.status === PlaylistStatus.PUBLISHED || playlist.userId === userId) {
        return next();
      }
```
In case of an error, this error will be thrown at the previous level.